kernel/comps/network/src/lib.rs

```diff
@@ -148,33 +170,32 @@ pub fn all_devices() -> Vec<(String, NetworkDeviceRef)> {
 }
 
 static COMPONENT: Once<Component> = Once::new();
-pub(crate) static NETWORK_IRQ_HANDLERS: Once<
-    SpinLock<Vec<Arc<dyn NetDeviceIrqHandler>>, LocalIrqDisabled>,
-> = Once::new();
 
 #[init_component]
 fn init() -> Result<(), ComponentInitError> {
     let a = Component::init()?;
     COMPONENT.call_once(|| a);
-    NETWORK_IRQ_HANDLERS.call_once(|| SpinLock::new(Vec::new()));
+    SoftIrqLine::get(NETWORK_TX_SOFTIRQ_ID).enable(handle_tx_softirq);
+    SoftIrqLine::get(NETWORK_RX_SOFTIRQ_ID).enable(handle_rx_softirq);
     buffer::init();
     Ok(())
 }
 
-type NetDeviceIrqHandlerListRef =
-    Arc<SpinLock<Vec<Arc<dyn NetDeviceIrqHandler>>, LocalIrqDisabled>>;
-type NetworkDeviceRef = Arc<SpinLock<dyn AnyNetworkDevice, LocalIrqDisabled>>;
+type NetDevicePollCallbackListRef =
+    Arc<SpinLock<Vec<Arc<dyn NetDevicePollCallback>>, BottomHalfDisabled>>;
```

Comment:
`comps/network` does not seem to know about `Poll`, so how about removing the `Poll` keyword from the name?